### DISCUSSION TITLE
Use a custom test reporter instead of JSON reporter

### DISCUSSION THREAD

**Discussion by AlmarAubel** [vitest-dev/vscode#25] [Ideas]
With the [ new feature of Vitest](https://github.com/vitest-dev/vitest/pull/1136) you could pass a custom reporter to Vitest. What do you think, if we would implement a custom reporter which could directly interact with the vscode api. Instead of using the json reporter. I think this will also make it easier to implement the watch function.

Otherwise we could directly interact with Vitest api (instead of using the CLI) and inject a reporter instance. 

For example
`
const vitestPath = getVitestPath(); // could return something like "/Users/xxx/source/vitest-explorer/samples/basic/node_modules/vitest/dist/node.js"
 const { startVitest } = await import(vitestPath); // Dynamic import during runtime using the Vitest instance of the user

      await startVitest(["basic.test.ts"], {
        watch: false,
        testNamePattern: testNamePattern,
        root: "/Users/xxxx/source/vitest-explorer/samples/basic",
        reporters: new VsCodeReporter(),
      });`
@zxch3n what do you think?

**Comment by zxch3n** [comment]
I think the best way to do this is to use the same interface as `@vitest/ui` does. So maybe we should use an approach like the second way. I'll investigate what's the best way to do this. It may require some changes on the vitest repo

**Reply by AlmarAubel** [reply]
Ok please let me know. Would love to help implementing this festure
